addons/website_sale/data/data.xml

```diff
@@ -96,14 +96,7 @@ for product in products:
 response = res_products
             </field>
         </record>
-        <!-- Dynamic Filter -->
-        <record id="dynamic_filter_demo_products" model="website.snippet.filter">
```

Comment:
You can't delete records in stable, people might have code that relies on this

Probably add something like this in the post_init

```python
filter = self.env.ref('website_sale.dynamic_filter_demo_products', False)
if action and filter:
        for website in env['website'].search([('id', '!=', filter.website_id.id)]):
```